### ISSUE TITLE
[expo-linear-gradient][Android] Linear-gradient rendering incorrectly on Android 13

### ISSUE THREAD

**Issue by joelodd** [expo/expo#24424]
### Minimal reproducible example

https://github.com/joelodd/expo-linear-gradient-android

### Summary

In the current version gradients simply aren't working on Android.

This happens and can be reproduced on the snack available via https://docs.expo.dev/versions/latest/sdk/linear-gradient/ 

iOS version
<img width="328" alt="Screenshot 2023-09-13 at 12 17 46" src="https://github.com/expo/expo/assets/129503008/b5c87364-a036-4971-a909-1bbe5ea5106b">

Android version
<img width="329" alt="Screenshot 2023-09-13 at 12 12 35" src="https://github.com/expo/expo/assets/129503008/a7de70df-451c-4eff-bd05-06757ef4e8ce">

### Environment

expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.3
      Shell: 3.6.0 - /opt/homebrew/bin/fish
    Binaries:
      Node: 18.4.0 - ~/.nvm/versions/node/v18.4.0/bin/node
      Yarn: 1.22.19 - /opt/homebrew/bin/yarn
      npm: 8.12.1 - ~/.nvm/versions/node/v18.4.0/bin/npm
    Managers:
      CocoaPods: 1.11.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
      Android SDK:
        API Levels: 31, 32, 33
        Build Tools: 30.0.3, 32.0.0, 33.0.0, 33.0.0
        System Images: android-33 | Google Play ARM 64 v8a
    IDEs:
      Android Studio: 2021.2 AI-212.5712.43.2112.8815526
      Xcode: 14.3.1/14E300c - /usr/bin/xcodebuild
    npmPackages:
      expo: ~49.0.5 => 49.0.10 
      react: 18.2.0 => 18.2.0 
      react-native: 0.72.4 => 0.72.4 
    npmGlobalPackages:
      eas-cli: 3.17.1
      expo-cli: 6.3.10
    Expo Workflow: managed

**Comment by brentvatne** [comment]
i can't reproduce this on snack by following the link from the docs:

<img width="1470" alt="image" src="https://github.com/expo/expo/assets/90494/bc98b61d-4fb3-4649-a9a7-728becd696b6">

**Comment by joelodd** [comment]
I am sorry @brentvatne but I can reproduce this everytime and even with my device (Google Pixel 4A - Android 13) connected to Expo from Snack.

My device - Google Pixel 4a, Android 13
![photo_2023-09-14 10 49 36](https://github.com/expo/expo/assets/129503008/e75d2198-35f9-4d80-910c-05d72a80f3c0)

On snack in MacOS - Google Chrome Version 116.0.5845.140 (Official Build) (arm64)
![Screenshot 2023-09-14 at 10 46 08](https://github.com/expo/expo/assets/129503008/05856ed1-edb1-4044-833b-0d022b9b57bd)

On snack in MacOS - Mozilla Firefox 117.0 (64-bit)
![Screenshot 2023-09-14 at 10 50 08](https://github.com/expo/expo/assets/129503008/0b14ff6f-c9a2-41ed-931f-37a4eac38252)

Can you please advise on further testing?
If I am doing something wrong please let me know as this is very weird.

**Comment by brentvatne** [comment]
related issue on our sister library: https://github.com/react-native-linear-gradient/react-native-linear-gradient/issues/628

**Comment by Kudo** [comment]
thanks for reporting the issue. it should be resolved now.
please also note that expo-go would cache some downloaded dependencies locally. before trying it, please clear expo-go's app data first.

**Comment by joelodd** [comment]
Hello @Kudo, Thank you for your time.

Unfortunately, I still can reproduce the issue on the [Snack for LinearGradient](https://snack.expo.dev/@joelclaudio/linear-gradient) on android (as seen on the screenshot below) and on my Google Pixel 4A, even after clearing the cache on Expo Go.

Please keep in mind that this looks like an issue with expo-linear-gradient itself and not the snack platform.

I've detected this on my own app where instead of rendering a gradient transparent-black to black overlay of an image, it simply renders a full black layer.
I've only mentioned the snack in the issue as it's the most quick way to reproduce the issue.

Can you please help? Am I missing something?

![Screenshot 2023-10-03 at 11 55 10](https://github.com/expo/expo/assets/129503008/962ce35f-d324-4634-af25-389d3ab706e9)

**Comment by Kudo** [comment]
@joelodd clearing cache is not enough because the file is stored in file system. could you try to use the "Clear Storage" for Expo Go in Android settings?

for the [example you provided](https://github.com/joelodd/expo-linear-gradient-android), i cannot really reproduce the problem. would be good if you could share how you test the example, for instance, by `npx expo start` or `npx expo run:android`. thanks!

**Comment by joelodd** [comment]
Hello @Kudo,
Sorry for taking so long to reply. I've finally managed to resolve my issue. It was associated with using expo-image and expo-linear-gradient.

Although it did rendered correctly on iOS, on Android it would simply display a black box instead of a nice "black to transparent gradient" overlay on top of an image.

For others facing my issue, the problem and fix was easy.

The problem, having <LinearGradient> as a child of <Image> like so:
`<Image>
     <LinearGradient />
</Image>`

The solution, as described [here](https://stackoverflow.com/questions/70588017/image-fades-into-background-react-native) was to simply put those two side by side, like so:
`<View>
    <Image />
    <LinearGradient />
</View>`

Thanks for everyone's time and effort.
I just thought that this was a more known issue when I saw the error on the Snack site.

Best Regards
